### PR TITLE
Update README and contribution guidelines

### PR DESCRIPTION
**Pull request by HCastano**
Just some small improvements in anticipation of onboarding new developers.

**README**
There were a few links I wanted to add here. I also moved some of the ink! attributes
around to make the most common ones appear near the top.

[Rendered](https://github.com/paritytech/ink/blob/hc-update-readme-and-contribution-guidelines/README.md)

**CONTRIBUTING**
This hasn't been touched in years. I've updated it a bit more to reflect our current
"process", but it could still use a bit of love imo.

[Rendered](https://github.com/paritytech/ink/blob/hc-update-readme-and-contribution-guidelines/CONTRIBUTING.md)

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
## 🦑 📈 ink! Example Contracts ‒ Changes Report 📉 🦑
 ⚠️ **The ink! `master` is ahead of your branch, this might skew the comparison data below.**

These are the results when building the `examples/*` contracts from this branch with `cargo-contract 0.17.0-10640bc` and comparing them to ink! `master`: 

|                          | Δ Optimized Size | Δ Used Gas | Total Optimized Size | Total Used Gas |
|:--------------------------|------------------:|------------:|----------------------:|----------------:|
| `accumulator`            |                  |            | 1.13 K               |                |
| `adder`                  |                  |            | 2.18 K               |                |
| `contract-introspection` |                  |            | 2.39 K               |                |
| `contract-terminate`     |                  |            | 0.94 K               | 214_418        |
| `contract-transfer`      |                  |            | 8.15 K               | 14_418         |
| `delegator`              |                  |            | 6.12 K               | 46_616         |
| `dns`                    |                  |            | 9.39 K               | 43_254         |
| `erc1155`                |                  |            | 27.23 K              | 86_508         |
| `erc20`                  |                  |            | 8.95 K               | 43_254         |
| `erc721`                 |                  |            | 13.27 K              | 115_344        |
| `flipper`                |                  |            | 1.41 K               | 14_418         |
| `incrementer`            |                  |            | 1.30 K               | 14_418         |
| `multisig`               |                  |            | 25.87 K              | 93_984         |
| `proxy`                  |                  |            | 3.06 K               | 29_586         |
| `rand-extension`         |                  |            | 4.21 K               | 14_418         |
| `subber`                 |                  |            | 2.20 K               |                |
| `trait-erc20`            |                  |            | 9.23 K               | 43_254         |
| `trait-flipper`          |                  |            | 1.10 K               | 14_418         |
| `trait-incrementer`      |                  |            | 1.27 K               | 28_836         |
| `upgradeable-contract`   |                  |            | 3.11 K               |                |



[Link to the run](https://gitlab.parity.io/parity/ink-waterfall/-/pipelines/182951) | Last update: Fri Mar 11 04:51:33 CET 2022

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,26 +24,20 @@ You can find our code of conduct [here](CODE_OF_CONDUCT.md).
...
-    - `[env]` For changes to `ink_env`.
-    - `[lang]` For changes to `ink_lang`.
-    - `[allocator]` For changes to `ink_allocator`.
-    - `[metadata]` For changes to `ink_metadata`.
-    - `[prelude]` For changes to `ink_prelude`.
-    - `[primitives]` For changes to `ink_primitives`.
-    - `[ci]` For changes to the continuous integration.
-    - `[chores]` For general non-technical changes.
-    - `[examples]` For changes to the examples.
+We'd appreciate it if kept your commits small and descriptive. It makes the review
```
Comment (cmichi) [inline CONTRIBUTING.md:30]:
```suggestion
But please keep your commits small and descriptive. A good rule of thumb is to 
bundle semantic changes together in one commit. It makes the review
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -13,26 +24,20 @@ You can find our code of conduct [here](CODE_OF_CONDUCT.md).
...
 
-For other purposes and sections please try to find a good fit for the topic.
+To help you out here's a [really good post](https://cbea.ms/git-commit/) on how to write good commit
+messages.
 
 ### Checklist
 
 Below is a checklist for you before doing a pull request.
+
 Following these will ensure that your pull request is going to be accepted.
```
Comment (cmichi) [inline CONTRIBUTING.md:41]: Wow this checklist below is super exhaustive, I don't think anybody will ever go through the whole thing. We should simplify it as well. 

There are two `scripts/check-*` scripts to do a number of these things of this locally, could you check if they are referenced somewhere in here and still working? We could refer to them instead.

Comment (cmichi) [reply CONTRIBUTING.md:41]: Let's do it in a follow-up.

**Review by cmichi — APPROVED** [review]

**Pull request merged**